### ISSUE TITLE
[data grid] Replace ExcelJS

### ISSUE THREAD

**Issue by joserodolfofreitas** [mui/mui-x#13267]
## Motivation

- ExcelJS is not actively maintained and has many issues
    - [Security vulnerability coming from ExcelJS](https://github.com/mui/mui-x/issues/13187)
    - [Fix 'unsafe-eval' CSP issue](https://github.com/exceljs/exceljs/pull/2672) 
    - [Replace unzipper](https://github.com/exceljs/exceljs/pull/2687)

## Alternatives

- Fork ExcelJS and keep "external dependencies"
- Replace external dependency with internal code

**Comment by michelengelen** [comment]
Thanks @joserodolfofreitas 👍🏼

**Comment by oliviertassinari** [comment]
For what it's worth https://github.com/exceljs/exceljs/issues/2764

<img width="350" alt="SCR-20240529-chdk" src="https://github.com/mui/mui-x/assets/3165635/37de9f2a-4e8f-42c6-9d03-4715e1a1c17a">

See https://app.ashbyhq.com/home/upcoming/right-side/candidates/b3ff8b82-8c29-4aed-810b-1bede8f8d0e7/applications/0338b675-320f-480e-8b86-d591caabe9ef/feed

**Comment by oliviertassinari** [comment]
> Use alternative library https://github.com/SheetJS/sheetjs

Exceljs is part of our public API. It could be a breaking change.

**Comment by vs-sfalsig** [comment]
Any progress on this? 

Apart from the potential security issues, ExcelJS in its current form seems to also pull in [buffers 0.1.1](https://www.npmjs.com/package/buffers) (through the outdated unzipper -> binary dependencies), which, from what I can see, doesn't come with any license. Best case, that's annoying for our license audit tools which flags it and requires us to handle it as an exception; worst case, it's a legal risk as there's nothing explicitly allowing us to use that code.

**Comment by MathisPct** [comment]
Are you plan to publish the npm package `@mui/x-internal-exceljs-fork` as public package ?

**Comment by cherniavskii** [comment]
@MathisPct It already is a [public NPM package](https://www.npmjs.com/package/@mui/x-internal-exceljs-fork) and it replaced the `exceljs` dependency of the Data Grid Premium.
It is intended for Data Grid's internal usage though.

**Comment by pechersky** [comment]
The new fork triggers the following vite warning:
```
[EVAL] Warning: Use of `eval` function is strongly discouraged as it poses security risks and may cause issues with minification.
 - Use of `eval` function here. in node_modules/@mui/x-internal-exceljs-fork/dist/exceljs.min.js at 1038491..1038495
```

**Comment by cherniavskii** [comment]
@pechersky Is this a CSP error in the browser, or something else?

**Comment by pechersky** [comment]
No, this is a command line info output when doing `vite build` after it lists all the chunks it has generated.

**Comment by cherniavskii** [comment]
OK, I can reproduce it.

I thought it was only CSP that complained about `eval`, but I couldn't reproduce it in https://github.com/mui/mui-x/issues/11153#issuecomment-3363030120 and therefore did not remove the `eval`.

Just to confirm – is this a non-blocking warning that doesn't cause any issues?

**Comment by pechersky** [comment]
Correct, nonblocking , that's why I left it as a comment here instead of a new issue.
